### PR TITLE
AMP changes

### PR DESCRIPTION
**Pull request by wichert**
I'm afraid this PR is a bit messy. At this point I am mostly hoping to get an indication of interest for the changes in here. If there is interest we can work on cleaning things up and creating a a set of smaller PRs for individual changes.

This PR is a collection of changes that have been made by the AMP team in Woven Planet to help us configure geoserver clusters. The changes include:

- support for additional endpoints
- a couple of bugfixes
- some code cleanup (gitignore, use stdlib http status, use of go modules
- simpify the docker container

### CODE REVIEW

**Comment by hishamkaram** [comment]
@wichert thank you for your contribution, i will take a look on your Pull Request

File: datastores.go [hunk 1]
```diff
@@ -43,13 +44,16 @@ type DatastoreDetails struct {
...
-	Type   string
+	Name           string
+	Host           string
+	Port           int
+	DBName         string
+	DBUser         string
+	DBPass         string
+	Type           string
+	Schema         string
+	MinConnections int
```
Comment (eknowles) [inline datastores.go:55]: if you want more here are the ones I put in the options

```
{
				"Evictor run periodicity",
				"300",
			},
			{
				"Max open prepared statements",
				"50",
			},
			{
				"encode functions",
				"true",
			},
			{
				"Batch insert size",
				"1",
			},
			{
				"preparedStatements",
				"false",
			},
			{
				"Loose bbox",
				"true",
			},
			{
				"SSL mode",
				"DISABLE",
			},
			{
				"Estimated extends",
				"true",
			},
			{
				"fetch size",
				"1000",
			},
			{
				"Expose primary keys",
				"false",
			},
			{
				"validate connections",
				"true",
			},
			{
				"Support on the fly geometry simplification",
				"true",
			},
			{
				"Connection timeout",
				"20",
			},
			{
				"create database",
				"false",
			},
			{
				"min connections",
				"1",
			},
			{
				"max connections",
				"10",
			},
			{
				"Evictor tests per run",
				"3",
			},
			{
				"Test while idle",
				"true",
			},
			{
				"Max connection idle time",
				"300",
			},
```

**Comment by eknowles** [comment]
thank you for your contribution, what are your plans for the logger @wichert ?

**Comment by wichert** [comment]
> thank you for your contribution, what are your plans for the logger @wichert ?

Are you referring to the changes in 84727c1ca7dbdb1acfe799053110ab88dae59f87 ? I suspect that was done to decrease verbosity and move logging control to the caller.

**Comment by hishamkaram** [comment]
@wichert 
- could you update your branch and check if the unit test passes?  (I had to revert this PR [#15](https://github.com/hishamkaram/geoserver/pull/15) because the unit test is failing)
- I would like to know which versions of geoserver this PR is compatible with?

**Comment by hishamkaram** [comment]
Hey @wichert — picking this up four years later. 👋

Reviewed your PR as part of the v1.1 revival, and the feature-type discovery endpoint you added (`GetFeatureTypeList` against `?list=available|configured|all`) just landed on `master` in #38 (commit $(git rev-parse --short HEAD)). It's been refactored to fit the current package idiom — `*Context` twin, typed `FeatureTypeListKind` enum (`FeatureTypeListConfigured` / `FeatureTypeListAvailable` / `FeatureTypeListAvailableWithGeom` / `FeatureTypeListAll`), parallel entry in the `FeatureTypeServiceWithContext` interface — but the design and the response-shape modeling (`{"list":{"string":[...]}}`) are yours.

Combined with `CreateFeatureType` (also recently landed via #37), this gives a complete discover-and-publish workflow:

```go
available, _ := gs.GetFeatureTypeList(ws, ds, geoserver.FeatureTypeListAvailable)
for _, table := range available {
    _, _ = gs.CreateFeatureType(ws, ds, &geoserver.FeatureType{Name: table, NativeName: table, Srs: "EPSG:4326"})
}
```

The other changes in your PR were already covered elsewhere in the v1.1 revival — stdlib `http.MethodGet`/`StatusOK`, go modules, GitHub Actions in place of Travis, modernized Docker, slog-based `*Logger` wrapper, and an `Options []Entry` field on `DatastoreConnection` (from PR #37) that subsumes your `Schema` / `MinConnections` / `MaxConnections` typed fields. So the discovery endpoint was the unique contribution to bring forward.

Verified end-to-end against both GeoServer 2.27.4 LTS and 2.28.0 stable in our integration matrix; the seeded PostGIS table appears in `available` and moves to `configured` after publish.

Thank you for the original work and apologies for the long wait!

**Pull request closed without merging**